### PR TITLE
Implement SE-0039 (Modernizing Playground Literals)

### PR DESCRIPTION
**Pull request by tkremenek**
Implements SE-0039 (Modernizing Playground Literals).

### CODE REVIEW

File: include/swift/Parse/Tokens.def [hunk 1]
```diff
@@ -173,9 +187,19 @@ POUND_KEYWORD(endif)
...
 POUND_KEYWORD(sourceLocation)
-POUND_KEYWORD(available)
 POUND_KEYWORD(selector)
 
+// Keywords in the # namespace that are build configurations.
+POUND_CONFIG(available)
+
+// FIXME: It feels like a layering violation to mention the protocol
+// kind and description here, but essentially there is a single
+// delcarative way to specify the shape of object literals, their
```
Comment (lattner) [inline include/swift/Parse/Tokens.def]: typo "declarative".  

I think this approach is reasonable, no need for the FIXME.

**Comment by lattner** [comment]
Aside from the one comment typo, the patch looks great!  Question though: what is the migration plan for this?  This isn't producing fixits from the old to new syntax, and we want to make sure that playgrounds which use these get moved forward.  It is probably worth it to keep the lexer/parser support for the old syntax in just to provide support to move to the new syntax.

**Comment by tkremenek** [comment]
I was thinking about the migration plan as well,  and that we should have fix-its.   I agree keeping the old parser support --- or reworking it for that use case --- makes that possible.

**Comment by tkremenek** [comment]
@swift-ci test

**Comment by tkremenek** [comment]
The build failure is for building for iOS:

~~~
swift/stdlib/public/SDK/UIKit/UIKit.swift:222:31: error: initializer 'init(red:green:blue:alpha:)' with Objective-C selector 'initWithRed:green:blue:alpha:' conflicts with previous declaration with the same Objective-C selector
  public required convenience init(red: Float, green: Float,
                              ^
UIKit.UIColor:31:12: note: 'init' previously declared here
    public init(red: CGFloat, green: CGFloat, blue: CGFloat, alpha: CGFloat)
           ^
~~~

The problem is that the `_ColorLiteralConvertible` protocol is defined as follows:
~~~
public protocol _ColorLiteralConvertible {
  init(red: Float, green: Float, blue: Float, alpha: Float)
}
~~~

Thus the **UIKit** overlay already provides an initializer with the same Objective-C selector.  We could define it that the protocol always requires `CGFloat`, or the initializer required by the protocol could be marked `@nonobjc`.

**Comment by tkremenek** [comment]
@swift-ci test

**Comment by tkremenek** [comment]
@swift-ci test

**Pull request merged**